trie.go

```diff
@@ -18,6 +21,7 @@ type Node struct {
 }
 
 type Trie struct {
+	sync.Mutex
```

Comment:
I'd rather have the mutex as a named field rather than embedding it in the Trie struct.

Reply:
Done